### ISSUE TITLE
[HOLD for payment 2023-05-29] Implement UI changes for managing 2FA in newdot 

### ISSUE THREAD

**Issue by MonilBhavsar** [Expensify/App#18080]
Add pages and components for user to manage 2FA from Newdot

1. Add new Routes https://docs.google.com/document/d/11C65_BCS6D_V08JKX_mftI5ocWTYcazfblunrh9QGek/edit#bookmark=id.6u7oj99x2zog
2. Build disable 2FA page https://docs.google.com/document/d/11C65_BCS6D_V08JKX_mftI5ocWTYcazfblunrh9QGek/edit#bookmark=id.ld2a60lrxlj6
3. Build 2FA codes page content https://docs.google.com/document/d/11C65_BCS6D_V08JKX_mftI5ocWTYcazfblunrh9QGek/edit#bookmark=id.evab2rvg00as
4. Build verify 2FA page https://docs.google.com/document/d/11C65_BCS6D_V08JKX_mftI5ocWTYcazfblunrh9QGek/edit#bookmark=id.vm6ue9ss3aaw
5. Build 2FA success page https://docs.google.com/document/d/11C65_BCS6D_V08JKX_mftI5ocWTYcazfblunrh9QGek/edit#bookmark=id.4kfqwc56mdwm
6. Finally add an option is security settings page https://docs.google.com/document/d/11C65_BCS6D_V08JKX_mftI5ocWTYcazfblunrh9QGek/edit#bookmark=id.jz73dunvl4v0

Most of the code is written in the doc

---

Main tracking issue https://github.com/Expensify/App/issues/15215

**Comment by MonilBhavsar** [comment]
@thiagobrez sorry looks like I assigned you to the wrong issue.
Could you please comment here so that I can assign you here

**Comment by thiagobrez** [comment]
No worries @MonilBhavsar !

**Comment by MonilBhavsar** [comment]
Thanks! We're almost unblocked. Feel free to go through the doc and you can start working on it. Always here to help if you have any questions. Thank you!

**Comment by thiagobrez** [comment]
@MonilBhavsar Awesome, thank you very much 🙇🏻  Could you give me permissions to access the doc?

**Comment by bfitzexpensify** [comment]
Internal handling happening via https://github.com/Expensify/App/issues/15215, so don't think I'm needed here. Unassigning myself

**Comment by thiagobrez** [comment]
@MonilBhavsar Yes! It worked and I'm already on it, thank you very much! Also, I see that the figma designs are in the works and expected to be done by Monday. Can I have access to that also when it's done? Thank you :)

**Comment by MonilBhavsar** [comment]
We have finalized the design and final designs are updated in the doc. Is that fine, or you need designs exported directly from Figma?

**Comment by thiagobrez** [comment]
I think I was expecting to have access to the actual design file so I could measure and do everything pixel-perfect. But if that is not possible no worries, I guess with the pre-made components and detailed implementation in the doc it is already possible

**Comment by thiagobrez** [comment]
Update: Good progress on UI. Enabled, Disabled and Codes pages are done, Verify page is almost. After that, Success page should be quick.

**Comment by thiagobrez** [comment]
Update: UI is done and connected to API. Next up is testing on all platforms and writing tests.

**Comment by MonilBhavsar** [comment]
@shawnborton, @thiagobrez is working on implementation of 2FA settings and requires a SVG/animation file

We have this svg/animation on the success page of the 2FA settings.

<img width="388" alt="Screenshot 2023-05-08 at 3 49 35 PM" src="https://user-images.githubusercontent.com/32012005/236802117-ece0bbda-e94d-43c0-9a71-915007b7bda7.png">

This looks fireworks and we also have fireworks in the success page of Bank Accounts page. That looks like this (plugged in 2FA settings page) - 


https://user-images.githubusercontent.com/32012005/236802355-c5d85a8d-2e5d-4fdc-8840-b719d96953de.mov

Do we want to use the same fireworks animation or different one as proposed in Figma?
Also when exporting that SVG from Figma, it shows colorless in App and looks like we export from Adobe. If so could you please help us to export that file. Thanks! 🙏

**Comment by shawnborton** [comment]
Can you show me what is proposed in Figma that might be different from the animation you linked above?

Personally I think we should just use the same fireworks animation we're using elsewhere.

**Comment by shawnborton** [comment]
Oh, the screenshot you shared is basically the same as the animation so we're all good.

**Comment by thiagobrez** [comment]
Perfect, thank you both @shawnborton @MonilBhavsar !

**Comment by thiagobrez** [comment]
Update: Draft PR is being reviewed internally

**Comment by thiagobrez** [comment]
Hey @shawnborton! Is this security settings page header expected to be implemented in this PR? Asking because I did not see any mention for it in the design doc:

<img width="382" alt="Screenshot 2023-05-10 at 10 14 38" src="https://github.com/Expensify/App/assets/26878038/460c480b-d055-40fc-9c43-b331b09f0dae">

cc @MonilBhavsar

**Comment by MonilBhavsar** [comment]
Good question! I don't think so. I guess this should be a part of another PR where we apply header images to all pages, so we have consistency in design. But yeah, @shawnborton would be knowing better.

**Comment by shawnborton** [comment]
Yeah, we'll handle those design changes in a separate PR so you can ignore them for now!

**Comment by thiagobrez** [comment]
PR is here and ready for review!

https://github.com/Expensify/App/pull/18576

**Comment by MonilBhavsar** [comment]
Thank you!

**Comment by MonilBhavsar** [comment]
@thiagobrez the QR code is being added [here](https://github.com/Expensify/App/pull/18636) and with the updated branding.
If that PR is merged before us, we'll merge main in your PR and then use that QR code component. cc @shawnborton 
Looks like the Expensify logo we need in the middle of QR code is the same as the one at http://new.expensify.com/settings/about

<img width="1089" alt="QRcodes" src="https://github.com/Expensify/App/assets/32012005/ba21f34e-ceec-402f-9624-67b71d64e092">

**Comment by thiagobrez** [comment]
Perfect, this is gorgeous! 👍🏻

**Comment by MonilBhavsar** [comment]
@thiagobrez the linked PR is merged now. If you could please merge main in to your PR and use that new library

**Comment by thiagobrez** [comment]
@MonilBhavsar Done 👍🏻

Here is the result:

| Web | iOS | Android |
|--------|--------|--------|
| <img width="1332" alt="Screenshot 2023-05-15 at 11 01 27" src="https://github.com/Expensify/App/assets/26878038/1103f229-6ed6-4937-ae27-7b00ab3ae029"> | ![Simulator Screen Shot - iPhone 14 - 2023-05-15 at 10 53 03](https://github.com/Expensify/App/assets/26878038/6c574859-d192-472f-9802-fd55ca184cda) | ![Screenshot 2023-05-15 at 11 14 09](https://github.com/Expensify/App/assets/26878038/94b7e993-d20d-4259-b393-0d0a992aeb92) |

I didn't use the `title`, `subtitle` and `logo` props from the new component because it didn't seem appropriate here, but let me know if you think it should be added

**Comment by MonilBhavsar** [comment]
I think we might want to ungroup the QR code component from the QR share component.
The QR code component would only render a QR code and we can use that in QR share as well as in our 2FA page to render a QR code to achieve the results as the mockup above.

**Comment by thiagobrez** [comment]
@MonilBhavsar Created a reusable `QRCode` component that is being used by both `QRShare` and `VerifyPage`.

Result:
| Verify Page | Share code page |
|--------|--------|
| ![Simulator Screen Shot - iPhone 14 - 2023-05-15 at 12 01 03](https://github.com/Expensify/App/assets/26878038/c0f1e217-9c8f-4adb-b55a-3a4a09d7052d) | ![Simulator Screen Shot - iPhone 14 - 2023-05-15 at 12 01 33](https://github.com/Expensify/App/assets/26878038/99ff178c-28c2-4de1-ac37-8d99c58ca8d1) |

**Comment by MonilBhavsar** [comment]
Thank you! Looks great.
In 2FA verify page, let's update background color to match the share QR code background color, and pass Expensify logo to make it look like the new mockup.

<img width="249" alt="Screenshot 2023-05-15 at 4 38 33 PM" src="https://github.com/Expensify/App/assets/32012005/b9f02a8a-05e4-4bb8-bbc9-9c9f2458d5e3">

**Comment by thiagobrez** [comment]
Similar to `src/assets/expensify-logo-round.png`, but inverted. 400x400px is good enough :)

Thank you in advance!

![Screenshot 2023-05-15 at 14 25 13](https://github.com/Expensify/App/assets/26878038/239a4ab7-4a4c-4635-a376-3ab34366862e)

**Comment by shawnborton** [comment]
In this screenshot here:
<img width="361" alt="image" src="https://github.com/Expensify/App/assets/2319350/d39f2003-f269-4a25-a920-315c6975c0cf">


For the email/login part, can we use our textSupporting color please? Thanks!

**Comment by thiagobrez** [comment]
>Happy to help provide any specific images/svgs you might need, just let me know!

Thanks @shawnborton!

I need the Expensify logo, with dark background and light fill color, in 400x400px, so I can put it in the middle of the 2-factor authentication QR Code.

The INVERSE of this one ⬇️ 

![image](https://github.com/Expensify/App/assets/26878038/31f7c66c-8452-4393-a1d2-93432675888b)

**Comment by thiagobrez** [comment]
>For the email/login part, can we use our textSupporting color please? Thanks!

Maybe this should be addressed in https://github.com/Expensify/App/pull/18636? This screenshot was just showing that there were no regressions in the Share Code feature. I haven't touched that in this issue.

But I know the Share Code PR is already closed, so if you want me to fix it in this PR anyway, I can do it easily. Just wanted to give the heads up :D

cc @MonilBhavsar

**Comment by MonilBhavsar** [comment]
Agree with @thiagobrez here.
We need to open another issue and PR to address this concern.

**Comment by shawnborton** [comment]
That works for me! Can you make sure to follow up on the respective issue (or create a new one) for that?

**Comment by MonilBhavsar** [comment]
Sure, looks like there have been some bug reports for QR code already. If we can spin up a PR and polish them all? cc @robertjchen

**Comment by robertjchen** [comment]
Yep! We're working on the bug reports for the QR code here: https://github.com/Expensify/App/issues/18968

**Comment by MonilBhavsar** [comment]
That's awesome! 🙌  
It would be great to fix the issue Shawn mentioned [here](https://github.com/Expensify/App/issues/18080#issuecomment-1548179869) in that PR.

**Comment by thiagobrez** [comment]
> @thiagobrez which format do you need? svg or png?

@shawnborton PNG, please 🙌🏻

**Comment by shawnborton** [comment]
Okay, curious how we'll handle this one for light mode too cc @grgia something to consider here. 

Here is the png: [appicon_qrcode.png.zip](https://github.com/Expensify/App/files/11485538/appicon_qrcode.png.zip)

**Comment by grgia** [comment]
Added to light mode doc to track, thanks!

**Comment by thiagobrez** [comment]
PR is merged 🏁

**Comment by MonilBhavsar** [comment]
Not sure, why this wasn't closed by PR.
Closing this now. Thanks @thiagobrez!
